### PR TITLE
Fixing for Back-port of IDENTITY-7050

### PR DESCRIPTION
**Pull request by ruwanta**
## Purpose
Made small changes to allow the code run on 5.3.0

## Goals
allow the code run on 5.3.0

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthHandler.java [hunk 1]
```diff
@@ -111,15 +105,21 @@ private boolean isValidJWTClientAssertionRequest(String clientAssertionType, Str
...
 
     @Override
-    public boolean authenticateClient(OAuthTokenReqMessageContext tokReqMsgCtx)
-            throws IdentityOAuth2Exception {
+    public boolean authenticateClient(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+        boolean isValid = false;
         try {
             SignedJWT signedJWT = getSignedJWT(tokReqMsgCtx);
-            return jwtValidator.isValidToken(signedJWT);
+            isValid = jwtValidator.isValidToken(signedJWT);
```
Comment (GayanM) [inline component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthHandler.java]: if isValid is false can return at this point

File: component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthHandler.java [hunk 2]
```diff
@@ -34,15 +36,7 @@
...
-import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.DEFAULT_AUDIENCE;
-import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.DEFAULT_ENABLE_JTI_CACHE;
-import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.DEFAULT_ISSUER;
-import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.DEFAULT_VALIDITY_PERIOD_IN_MINUTES;
-import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.OAUTH_JWT_ASSERTION;
-import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.OAUTH_JWT_ASSERTION_TYPE;
-import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.OAUTH_JWT_BEARER_GRANT_TYPE;
-import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.PREVENT_TOKEN_REUSE;
-import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.REJECT_BEFORE_PERIOD;
+import static org.wso2.carbon.identity.oauth2.token.handler.clientauth.jwt.Constants.*;
```
Comment (GayanM) [inline component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthHandler.java]: can't expand imports ?

File: component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthHandler.java (hunk 1 above)
Comment (ruwanta) [reply component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthHandler.java]: +1, Fixed.

File: component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthHandler.java [hunk 3]
```diff
@@ -111,15 +105,21 @@ private boolean isValidJWTClientAssertionRequest(String clientAssertionType, Str
...
-            return jwtValidator.isValidToken(signedJWT);
+            isValid = jwtValidator.isValidToken(signedJWT);
+            if (isValid && StringUtils.isEmpty(tokReqMsgCtx.getOauth2AccessTokenReqDTO().getClientId())) {
+                ReadOnlyJWTClaimsSet claimsSet = jwtValidator.getClaimSet(signedJWT);
+                String jwtSubject = jwtValidator.resolveSubject(claimsSet);
+                tokReqMsgCtx.getOauth2AccessTokenReqDTO().setClientId(jwtSubject);
+            }
         } catch (IdentityOAuth2Exception e) {
-            //IdentityOAuth2Exception are logged when thrown
+            log.error("Error occurred validating the signed JWT", e);
```
Comment (GayanM) [inline component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthHandler.java]: log context information if available

**Pull request merged**